lib/elastic_apm.rb

```diff
@@ -285,27 +285,43 @@ def with_span(
     #
     # @param rack_env [Rack::Env] A Rack env
     # @return [Context] The built context
-    def build_context(rack_env)
-      agent&.build_context(rack_env)
+    def build_context(
+      deprecated_env = nil,
+      rack_env: nil,
+      for_type: :transaction
+    )
+      if !rack_env && (rack_env = deprecated_env)
```

Comment:
Why did you introduce `deprecated_env` here and make `rack_env` a keyword arg but then require it?
How about `def build_context(rack_env:, for_type:)` instead?

Reply:
Backwards compatibility. Used to be 1 non-keyword arg. To keep it working with a deprecation warning, I have both.